src/compile/data/transforms.ts

```diff
@@ -130,11 +132,24 @@ export function parseTransformArray(model: Model) {
   let previous: DataFlowNode;
   let lookupCounter = 0;
 
+  function insert(newNode: DataFlowNode) {
+    if (!first) {
+      // A parent may be inserted during node construction
+      // (e.g., selection FilterNodes may add a TimeUnitNode).
+      first = newNode.parent || newNode;
```

Comment:
I'm trying to find where this is happening. Why can we not references to the first and last node like the `parseTransformArray` method does?